test/integration/fake_upstream.cc

```diff
@@ -86,6 +86,12 @@ void FakeStream::encode1xxHeaders(const Http::ResponseHeaderMap& headers) {
   postToConnectionThread([this, headers_copy]() -> void {
     {
       absl::MutexLock lock(&lock_);
+      {
+        absl::MutexLock dtor_lock(&dtor_lock_);
```

Comment:
there's no way this can be the right answer.

The locking should be done *outside* the destructor, possibly based on whatever container is holding this object.

Reply:
I think most tests simply make sure work isn't going on as the upstreams are torn down.

Reply:
one pattern I've used in the past is to block the test-infrastucture destruction in the main test-method on all traffic being drained.